support-workers/src/typescript/lambdas/createZuoraSubscriptionTSLambda.ts

```diff
@@ -0,0 +1,144 @@
+import type { PaymentMethod as ZuoraPaymentMethod } from '@modules/zuora/orders/paymentMethods';
+import { ZuoraClient } from '@modules/zuora/zuoraClient';
+import type { Address } from '../model/address';
+import type { CreateZuoraSubscriptionState } from '../model/createZuoraSubscriptionState';
+import type { PaymentMethod } from '../model/paymentMethod';
+import { stageFromEnvironment } from '../model/stage';
+import type { WrappedState } from '../model/stateSchemas';
+import { ServiceProvider } from '../services/config';
+import { asRetryError } from '../util/errorHandler';
+import { getIfDefined } from '../util/nullAndUndefined';
+
+const stage = stageFromEnvironment();
+
+const zuoraServiceProvider = new ServiceProvider(stage, async (stage) => {
+	return ZuoraClient.create(stage);
+});
+
+const productCatalogProvider = new ServiceProvider(stage, async (stage) => {
+	return getProductCatalogFromApi(stage);
+});
+
+export const handler = async (
+	state: WrappedState<CreateZuoraSubscriptionState>,
+) => {
+	try {
+		console.info(`Input is ${JSON.stringify(state)}`);
+
+		const createZuoraSubscriptionState = state.state;
+		const productSpecificState =
+			createZuoraSubscriptionState.productSpecificState;
+
+		const user = createZuoraSubscriptionState.user;
+		const currency: IsoCurrency = createZuoraSubscriptionState.product.currency;
+		const zuoraPaymentMethod = getZuoraPaymentMethod(
+			productSpecificState.paymentMethod,
+		);
+
+		const billToContact: Contact = buildContact(
+			user.firstName,
+			user.lastName,
+			user.primaryEmailAddress,
+			user.billingAddress,
+		);
+
+		const salesforceContact = productSpecificState.salesForceContact;
+
+		const productInformation = getIfDefined(
+			productSpecificState.productInformation,
+			'productInformation is required',
+		);
+
+		// TODO:
+		//  Apply promotion if present
+		//  Validate paper payment gateway? Might be done already by schema
+		//  Output state
+		//  CSR mode is NOT needed
+
+		const inputFields: CreateSubscriptionInputFields<ZuoraPaymentMethod> = {
+			accountName: salesforceContact.AccountId, // We store the Salesforce Account id in the name field
+			createdRequestId: createZuoraSubscriptionState.requestId,
+			salesforceAccountId: salesforceContact.AccountId,
+			salesforceContactId: salesforceContact.Id,
+			identityId: createZuoraSubscriptionState.user.id,
+			currency: currency,
+			paymentGateway: productSpecificState.paymentMethod.PaymentGateway,
+			paymentMethod: zuoraPaymentMethod,
+			billToContact: billToContact,
+			productPurchase: productInformation,
+			runBilling: true,
+			collectPayment: true,
+		};
+
+		const zuoraClient = await zuoraServiceProvider.getServiceForUser(false);
+		const productCatalog = await productCatalogProvider.getServiceForUser(
+			createZuoraSubscriptionState.user.isTestUser,
+		);
+		const result = await createSubscription(
+			zuoraClient,
+			productCatalog,
+			inputFields,
+		);
+
+		// TODO: return the correct state
```

Comment:
I presume this TODO needs to be completed before this lambda could actually be used?